internal/store/store.go

```diff
@@ -230,6 +234,32 @@ func (s *lockingStore) DeleteRoute(ctx context.Context, id string) error {
 	return s.syncGateways(ctx, modifiedGateways)
 }
 
+// SyncAllAtInterval syncs all known Gateway(s) and Route(s) into Consul
+// at a regular interval. This function is blocking.
+func (s *lockingStore) SyncAllAtInterval(ctx context.Context) {
```

Comment:
This implements the background sync of all resources from the store added in #278 to deal with resources deleted out of band (such as a k8s Service, etc.)

Reply:
Assuming this will be covered by tests when it's actually consumed?

Reply:
This is already covered by an e2e test added in #278